build.psm1

```diff
@@ -432,7 +432,8 @@ Fix steps:
     # adding ErrorOnDuplicatePublishOutputFiles=false due to .NET SDk issue: https://github.com/dotnet/sdk/issues/15748
     # removing --no-restore due to .NET SDK issue: https://github.com/dotnet/sdk/issues/18999
     # $Arguments = @("publish","--no-restore","/property:GenerateFullPaths=true", "/property:ErrorOnDuplicatePublishOutputFiles=false")
-    $Arguments = @("publish","/property:GenerateFullPaths=true", "/property:ErrorOnDuplicatePublishOutputFiles=false")
+    # Add --self-contained due to "warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used."
+    $Arguments = @("publish","/property:GenerateFullPaths=true", "/property:ErrorOnDuplicatePublishOutputFiles=false", "--self-contained")
```

Comment:
This is going to be a change in the resulting build format going forward, isn't it? I don't think we've previously had self-contained executables for pwsh.

Reply:
the `--self-contained` switch was implicit when we specify --runtime. Now if moving to .NET 6 Preview 7 we get an error for that. `--self-contained` creates a package with the required .NET framework assemblies, where are `--no-self-contained` creates a framework dependent package.